src/periodic.jl

```diff
@@ -0,0 +1,85 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicParameter = Union{Real,Period}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::PeriodicFunction`: the periodic function
+* `period<:PeriodicParameter`: the duration it takes for the periodic function to repeat.
+    The sign of the value determines the direction of the function. Must be non-zero.
```

Comment:
```suggestion
```

these seem contradictory?

Reply:
Maybe you are thinking non-negative? That would be contradictory. I'm saying it can be negative or positive, but not equal to 0.

Reply:
yeah, non-negative, my mistake